### PR TITLE
LPS-190591 Manage Pages makes wrong pages visible on Page Tree

### PR DESCRIPTION
**Pull request by Mikellorza**
bug : https://liferay.atlassian.net/browse/LPS-190591

Fix description
----------
The method getLayoutsCount of LayoutServiceImple was not taking into account the MANAGE_LAYOUT permission, so, I have change it to check that permission.

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by Mikellorza** [comment]
ci:test:sf

File: modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/util/LayoutsTreeImpl.java [hunk 1]
```diff
@@ -204,8 +221,20 @@ private JSONArray _getLayoutsJSONArray(
...
+				themeDisplay.getPermissionChecker(),
+				themeDisplay.getScopeGroup(), ActionKeys.MANAGE_LAYOUTS)) {
+
+			count = _layoutLocalService.getLayoutsCount(
+				groupId, privateLayout, parentLayoutId);
+		}
+		else {
+			count = _layoutService.getLayoutsCount(
+				groupId, privateLayout, parentLayoutId);
+		}
```
Comment (lfbesada) [inline modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/util/LayoutsTreeImpl.java:237]: I think you forget to delete this, isn't it?

**Comment by Mikellorza** [comment]
@lfbesada I have changed the fix to check the permission MANAGE_LAYOUT in a centralized method and with that we can fix another bugs. I talk with you to create a method to return a list of plids to avoid get all layout information but in the end when we are using LayoutPermissionUtil with the plid I have realized that is getting the layout object, so, in the end that method to get a list with plids is not necessary.

**Comment by lfbesada** [comment]
ci:test:sf

**Review by lfbesada — APPROVED** [review]
LGTM 🎉

**Comment by lfbesada** [comment]
ci:forward

**Comment by Tim-Cao** [comment]
ci:forward:force

**Pull request closed without merging**